openedx/core/djangoapps/enrollments/serializers.py

```diff
@@ -83,15 +86,49 @@ class CourseEnrollmentSerializer(serializers.ModelSerializer):
 
     """
     course_details = CourseSerializer(source="course_overview")
-    user = serializers.SerializerMethodField('get_username')
+    user = serializers.SerializerMethodField("get_username")
+    finished = serializers.SerializerMethodField()
+    grading = serializers.SerializerMethodField()
 
     def get_username(self, model):
         """Retrieves the username from the associated model."""
         return model.username
 
-    class Meta:
+    def get_finished(self, model):
+        """Retrieve finished course."""
+        course = modulestore().get_course(model.course_id)
+        if course:
+            try:
+                coursegrade = CourseGradeFactory().read(model.user, course).passed
+            except PermissionDenied:
+                return False
+            return coursegrade
+        return False
+
+    def get_grading(self, model):
+        """Retrieve course grade."""
+        course = modulestore().get_course(model.course_id)
+        course_grade = None
+        summary = []
+        current_grade = 0
+        if course:
+            try:
+                course_grade = CourseGradeFactory().read(model.user, course)
+                current_grade = int(course_grade.percent * 100)
+                for section in course_grade.summary.get(u'section_breakdown'):
+                    if section.get(u'prominent'):
+                        summary.append(section)
+            except PermissionDenied:
+                pass
+        return [
+            {u'current_grade': current_grade,
+             u'certificate_eligible': course_grade.passed if course_grade else False,
```

Comment:
This is not consistent with the [certificates API](https://github.com/edx/edx-platform/blob/023f4144abca4a035b333f0e2a2de625b58f8c52/lms/djangoapps/certificates/api.py#L812-L816), so it can return different values than generated grade reports.

Reply:
Thanks for catching this @Agrendalath possibly one of the changes that have been impacting this PR since it has been created. I'll schedule some time to look into this, resolve conflicts and fix tests.